packit/schema.py

```diff
@@ -109,6 +110,38 @@ def validate_all_actions(self, actions: list) -> None:
             raise ValidationError(f"Unknown action(s) provided: {invalid_actions}")
 
 
+class SourcesItemField(fields.Field):
```

Comment:
Couldn't this be made a `SourceSchema`, inherit from `Schema` and referenced when typing the `sources` field?

Based on a quick scroll in this [docs](https://marshmallow.readthedocs.io/en/stable/nesting.html), it would look something like `fields.List(fields.Nested(SourceSchema))`.

I'm expecting that this way schema validation would "just work" and you could get rid of the code from `_deserialize()` as Marshmallow would handle it.

Though, I might be missing something...

Reply:
Okay, I see now that this is the pattern that is used for the rest of the fields.

Though the approach seems overly complicated to me. I would expect Marshmallow to handle all the type checking when deserializing the data.

(I have the feeling that this monologue is out of scope for this PR).

Reply:
I was following the pattern used before, but now I probably found a more simple way, will try it and update it.